galley/pkg/crd/validation/config.go

```diff
@@ -215,17 +257,161 @@ func rebuildWebhookConfigHelper(
+		caFile:                        p.CACertFile,
+		webhookConfigFile:             p.WebhookConfigFile,
+		clientset:                     p.Clientset,
+		webhookName:                   p.WebhookName,
+		deploymentName:                p.DeploymentName,
+		deploymentAndServiceNamespace: p.DeploymentAndServiceNamespace,
+		createInformerWebhookSource:   defaultCreateInformerWebhookSource,
+	}
+
+	if galleyDeployment, err := whc.clientset.AppsV1().Deployments(whc.deploymentAndServiceNamespace).Get(whc.deploymentName, v1.GetOptions{}); err != nil { // nolint: lll
+		scope.Warnf("Could not find %s/%s deployment to set ownerRef. The validatingwebhookconfiguration must be deleted manually",
+			whc.deploymentAndServiceNamespace, whc.deploymentName)
+	} else {
+		whc.ownerRefs = []v1.OwnerReference{
+			*v1.NewControllerRef(
+				galleyDeployment,
+				appsv1.SchemeGroupVersion.WithKind("Deployment"),
+			),
+		}
+	}
+
+	return whc, nil
+}
+
+//reconcile monitors the keycert and webhook configuration changes, rebuild and reconcile the configuration
+func (whc *WebhookConfig) reconcile(stopCh <-chan struct{}) {
+	defer whc.keyCertWatcher.Close() // nolint: errcheck
+	defer whc.configWatcher.Close()  // nolint: errcheck
+
+	// Try to create the initial webhook configuration (if it doesn't
+	// already exist). Setup a persistent monitor to reconcile the
+	// configuration if the observed configuration doesn't match
+	// the desired configuration.
+	if err := whc.rebuildWebhookConfig(); err == nil {
+		whc.createOrUpdateWebhookConfig()
+	}
+	webhookChangedCh := whc.monitorWebhookChanges(stopCh)
+
+	// use a timer to debounce file updates
+	var keyCertTimerC <-chan time.Time
+	var configTimerC <-chan time.Time
+
+	for {
+		select {
+		case <-keyCertTimerC:
+			keyCertTimerC = nil
+			whc.reloadKeyCert()
+		case <-configTimerC:
+			configTimerC = nil
+
+			// rebuild the desired configuration and reconcile with the
+			// existing configuration.
+			if err := whc.rebuildWebhookConfig(); err == nil {
+				whc.createOrUpdateWebhookConfig()
+			}
+		case <-webhookChangedCh:
+			// reconcile the desired configuration
+			whc.createOrUpdateWebhookConfig()
+		case event, more := <-whc.keyCertWatcher.Event:
+			if more && (event.IsModify() || event.IsCreate()) && keyCertTimerC == nil {
+				keyCertTimerC = time.After(watchDebounceDelay)
+			}
+		case event, more := <-whc.configWatcher.Event:
+			if more && (event.IsModify() || event.IsCreate()) && configTimerC == nil {
+				configTimerC = time.After(watchDebounceDelay)
+			}
+		case err := <-whc.keyCertWatcher.Error:
+			scope.Errorf("keyCertWatcher error: %v", err)
+		case err := <-whc.configWatcher.Error:
+			scope.Errorf("configWatcher error: %v", err)
+		case <-stopCh:
+			return
+		}
+	}
+}
+
+// ReconcileWebhookConfiguration reconciles the ValidatingWebhookConfiguration
+func ReconcileWebhookConfiguration(webhookServerReady chan struct{}, stopCh <-chan struct{}, kubeConfig string, vc *WebhookParameters) {
```

Comment:
Add a comment about webhookServerReady potentially being null. Also, both channel definitions can be of same type:
```
func ReconcileWebhookConfiguration(webhookServerReady, stopCh <-chan struct{}, ...)
```